scripts/prepare_sector_network.py

```diff
@@ -157,6 +157,113 @@ def add_generation(
         # set the "co2_emissions" of the carrier to 0, as emissions are accounted by link efficiency separately (efficiency to 'co2 atmosphere' bus)
         n.carriers.loc[carrier, "co2_emissions"] = 0
 
+    # add coal CC technologies
+    if options["coal"].get("CC", False):
+        coal_CC_techs = options["coal"].get("coal_CC_techs", list())
+
+        coal_CC_names = {
+            "coal CC-95": "Coal-95%-CCS",
+            "coal CC-99": "Coal-99%-CCS",
+            "coal IGCC CC-90": "Coal-IGCC-90%-CCS",
+        }
+
+        for cc_tech in coal_CC_techs:
+            n.madd(
+                "Link",
+                spatial.nodes + " " + cc_tech,
+                bus0=spatial.coal.nodes,
+                bus1=spatial.nodes,
+                bus2="co2 atmosphere",
+                bus3=spatial.co2.nodes,
+                marginal_cost=costs.at[coal_CC_names[cc_tech], "efficiency"]
+                * costs.at[coal_CC_names[cc_tech], "VOM"],  # NB: VOM is per MWel

```

Comment:
Thanks, that was actually a question that bothered me earlier, because in all other technologies are implemented that way. Therefore, I raised a question in PyPSA-Earth meeting. @davide-f explained the reason well. I think, yes, you are right. Marginal cost are for MWh in electrical side, therefore we multiply by efficiency.

Reply:
Perfect, we are on the same page then 🙂